From 6c37174d19256be207c7b68d85cf9dfe55eaff77 Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Wed, 8 Aug 2018 11:08:44 -0700
Subject: [PATCH 1598/1626] ASoC: SOF: update dai config for all dai with same
 name

Previously, dai config was updated only for one DAI with the same
name as the link name. But in the case of duplex pcm's, there will
be two DAI's with the same name and both of them need to be updated.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/topology.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index 44eae25..4f2e9cf 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -1832,17 +1832,17 @@ static int sof_link_load(struct snd_soc_component *scomp, int index,
 	if (ret < 0)
 		return ret;
 
-	dai = snd_sof_find_dai(sdev, (char *)link->name);
-	if (dai) {
-		/* set config for dai */
-		memcpy(&dai->dai_config, &config,
-		       sizeof(struct sof_ipc_dai_config));
-		return 0;
-	}
+	/* set config for all DAI's with name matching the link name */
+	list_for_each_entry(dai, &sdev->dai_list, list) {
+		if (!dai->name)
+			continue;
 
-	dev_err(sdev->dev, "failed to find BE DAI for link %s\n", link->name);
+		if (strcmp(link->name, dai->name) == 0)
+			memcpy(&dai->dai_config, &config,
+			       sizeof(struct sof_ipc_dai_config));
+	}
 
-	return -EINVAL;
+	return 0;
 }
 
 static int sof_link_unload(struct snd_soc_component *scomp,
-- 
2.7.4

